code/datums/extensions/health/atom.dm

```diff
@@ -0,0 +1,397 @@
+/atom
+	/**
+	 * Whether or not the atom should use the health handler when initialized.
+	 */
+	var/use_health_handler = FALSE
+
+	/**
+	 * Ease-of-use holder for the health extension.
+	 */
+	var/datum/extension/health/health_handler
+
+	/**
+	 * Current health for simple health processing
+	 */
+	var/simple_health
+
+	/**
+	 * Maximum health for simple health processing
+	 */
+	var/simple_max_health
+
+
+/// Returns a list containing the atom's requested health_handler configuration.
+/atom/proc/health_handler_config()
+	return list(
+		"max_health" = 100
+	)
+
+
+/**
+ * Initializes the health extension.
+ */
+/atom/proc/initialize_health()
+	var/list/health_handler_config = health_handler_config()
+
+	if (using_simple_health())
+		simple_max_health = health_handler_config["max_health"]
+		simple_health = simple_max_health
+		return
+
+	health_handler = get_or_create_extension(src, use_health_handler)
+	health_handler.set_max_health(health_handler_config["max_health"])
+
+	for (var/damage_type in DAMAGE_ALL)
+		if ("resist_[damage_type]" in health_handler_config)
+			set_damage_resistance(damage_type, health_handler_config["resist_[damage_type]"])
+
+	on_initialize_health(health_handler_config)
+
+
+/**
+ * To allow for additional init processing.
+ */
+/atom/proc/on_initialize_health(list/health_handler_config)
```

Comment:
I feel like you could just make `initialize_health()` return true/false and let people override it for additional logic.